compiler/luci/pass/src/FuseRmsNormPass.cpp

```diff
@@ -0,0 +1,216 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseRmsNormPass.h"
+#include "helpers/NodeFiller.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/Service/CircleNodeClone.h>
+
+#include <cmath>
+#include <cassert>
+
+namespace
+{
+
+/**
+ * Below diagram shows RMS normalization pattern to fuse.
+ * - this pattern will be replaced with one RmsNorm
+ *
+ *           [In]
+ *            |
+ *            V
+ *     +---- ifm ----+
+ *     |      |      |
+ *     |      V      |
+ *     |     mul <---+
+ *     |      |
+ *     |      V
+ *     |     mean
+ *     |      |
+ *     |      V
+ *     |     add_epsilon
+ *     |      |
+ *     |      V
+ *     |     rsqrt
+ *     |      |
+ *     |      V
+ *     +---> mul_input
+ *            |
+ *            V
+ *          [Out]
+ */
+
+class RmsNormPattern final
+{
+public:
+  RmsNormPattern(luci::CircleMul *candidate)
+  {
+    assert(candidate); // FIX_CALLER_UNLESS
+    _mul_input = candidate;
+  }
+
+public:
+  bool matched();
+
+public:
+  luci::CircleNode *_ifm = nullptr;
+  luci::CircleMul *_mul_pow = nullptr;
+  luci::CircleMean *_mean = nullptr;
+  luci::CircleAdd *_add_epsilon = nullptr;
+  luci::CircleRsqrt *_rsqrt = nullptr;
+  luci::CircleMul *_mul_input = nullptr;
+  luci::CircleConst *_const_epsilon = nullptr;
+  luci::CircleConst *_const_gamma = nullptr;
+  luci::CircleConst *_const_beta = nullptr;
+};
+
+#define CHECK_OR_FALSE(condition) \
+  if (not(condition))             \
+    return false;
+
+luci::CircleConst *make_const_one(loco::Graph *graph, float value)
+{
+  auto const_one = graph->nodes()->create<luci::CircleConst>();
+  const_one->dtype(loco::DataType::FLOAT32);
+  const_one->rank(1);
+  const_one->size<loco::DataType::FLOAT32>(1);
+  const_one->at<loco::DataType::FLOAT32>(0) = value;
+  return const_one;
+}
+
+bool RmsNormPattern::matched()
+{
+  CHECK_OR_FALSE(luci::fill(&_ifm, &_rsqrt).with_commutative_args_of(_mul_input));
+  _add_epsilon = dynamic_cast<luci::CircleAdd *>(_rsqrt->x());
+  CHECK_OR_FALSE(_add_epsilon);
+  CHECK_OR_FALSE(luci::fill(&_mean, &_const_epsilon).with_commutative_args_of(_add_epsilon));
+  CHECK_OR_FALSE(_const_epsilon->dtype() == loco::DataType::FLOAT32);
+  _mul_pow = dynamic_cast<luci::CircleMul *>(_mean->input());
+  CHECK_OR_FALSE(_mul_pow);
+  CHECK_OR_FALSE(_mul_pow->x() == _ifm);
+  CHECK_OR_FALSE(_mul_pow->y() == _ifm);
+
+  assert(_const_gamma == nullptr);
+  assert(_const_beta == nullptr);
+
+  /* NOTE: Current FuseRmsNormPass assumes no gamma(scale) and beta(bias).
+   But, RmsNorm kernel expects gamma and beta.
+   So, it creates default gamma(1.0) and beta(0.0).
+  */
```

Comment:
plz match comment begin/end style. current codes usually used the first one.
```suggestion
  /*
   NOTE: Current FuseRmsNormPass assumes no gamma(scale) and beta(bias).
   But, RmsNorm kernel expects gamma and beta.
   So, it creates default gamma(1.0) and beta(0.0).
  */
```
or
```suggestion
  /* NOTE: Current FuseRmsNormPass assumes no gamma(scale) and beta(bias).
   But, RmsNorm kernel expects gamma and beta.
   So, it creates default gamma(1.0) and beta(0.0). */
```

Reply:
Okay, I'll update it.